src/main/java/org/prebid/server/bidder/advangelists/AdvangelistsBidder.java

```diff
@@ -64,61 +64,64 @@ public AdvangelistsBidder(String endpointUrl) {
     }
 
     private static Map<ExtImpAdvangelists, List<Imp>> getImpToExtImp(BidRequest request, List<BidderError> errors) {
-        final List<Imp> imps = new ArrayList<>();
-        final List<ExtImpAdvangelists> extImpList = new ArrayList<>();
+        final Map<ExtImpAdvangelists, List<Imp>> parsedExtToListOfImp = new HashMap<>();
         for (Imp imp : request.getImp()) {
             try {
-                final ExtImpAdvangelists extImpEmxDigital = getExtImpAdvangelists(imp);
-                validateImpExt(extImpEmxDigital);
+                final ExtImpAdvangelists extImpEmxDigital = parseAndValidateImpExt(imp);
 
-                imps.add(imp);
-                extImpList.add(extImpEmxDigital);
+                parsedExtToListOfImp.putIfAbsent(extImpEmxDigital, new ArrayList<>());
+                parsedExtToListOfImp.get(extImpEmxDigital).add(imp);
```

Comment:
I think you can call updateImp here instantly. The only thing it will change - slightly modify an order in which an errors could be added, but that's not a critical thing, right @rpanchyk ?